src/Chess/Game/ChessGame.php

```diff
@@ -1367,12 +1365,58 @@ public function inClaimableDraw()
      */
     public function inRepetitionDraw()
     {
-        $fen = $this->renderFen(false);
-        if (isset($this->_allFENs[$fen]) && $this->_allFENs[$fen] >= 3) {
-            return true;
+        if (3 > $this->_halfMoves) {
+            return false;
         }
 
-        return false;
+        $samePositions = array_filter(
+            array_values(array_count_values($this->fens)),
+            function ($value) {
+                return 3 <= $value;
+            }
+        );
+
+        return (bool) $samePositions;
+    }
+
+
+    /**
+     * The game is drawn without either player making a claim if the same
+     * position occurs on the board after five consecutive pairs of moves
+     * for each player ("five-fold repetition")
+     *
+     * @return bool
+     */
+    public function inFiveFoldRepetitionDraw()
+    {
+        if (20 > $this->_halfMoves) {
+            return false;
+        }
+
+        $last5Moves = array_slice($this->fens, -20, 20);
```

Comment:
this seems like it is more like last 20 moves? or last 20 plys? any reason you called it last5moves?